agent/php_environment.c

```diff
@@ -511,6 +511,94 @@ static void nr_php_get_environment_variables(TSRMLS_D) {
                    __func__, NR_PHP_PROCESS_GLOBALS(env_labels));
 }
 
+#define DOCKER_ID_V2_STRLEN (64)
+/*
+ * Purpose:
+ *    Extract the 64-byte hexadecimal Docker cgroup ID from
+ *     /proc/self/mountinfo
+ */
+void nr_php_get_v2_docker_id(const char* cgroup_fname) {
+  char* line_ptr = NULL;
+  char* token = NULL;
+  bool found = false;
+  FILE* fd = NULL;
+  size_t len = 0;
+  nr_regex_t* regex = NULL;
+
+  // check if docker_id global already set
+  if (NULL != NR_PHP_PROCESS_GLOBALS(docker_id)) {
+    nrl_verbosedebug(NRL_AGENT, "Docker ID already set.");
+    return;
+  }
+
+  // check if file exists
+  if (SUCCESS != access(cgroup_fname, F_OK)) {
+    nrl_verbosedebug(NRL_AGENT, "File not found: %s", cgroup_fname);
+    return;
+  }
+
+  // open file
+  fd = fopen(cgroup_fname, "r");
+  if (NULL == fd) {
+    nrl_warning(NRL_AGENT, "Failed to open %s", cgroup_fname);
+    return;
+  }
+
+  // compile regex to verify hexadecimal chars
+  regex = nr_regex_create("^[a-fA-F0-9]+$", 0, 0);
```

Comment:
Can this call fail?

Reply:
It technically can if the pcre expression fails to compile (for example), in which case it returns `NULL`. The subsequent calls that depend on `regex` (`nr_regex_match` and `nr_regex_destroy`) are both capable of handling `NULL` values. That being said, it probably makes sense to bail out here if the regex fails for any reason.

Reply:
Addressed in 1e0ca9167b57752f376bf2237291f61568eaa144